### PR TITLE
ft(ZENKO-1616): Add fixes for new ObjectMD properties

### PR DESCRIPTION
**Pull request by tmacro**
Add special handling for the creation-time property

### CODE REVIEW

**Review by JianqinWang — APPROVED** [review]

**Review by ploki — COMMENTED** [review]
:+1: for the core logic. (we might not want to push the package.json stuff as-is though)

File: lib/api/apiUtils/object/createAndStoreObject.js [hunk 1]
```diff
@@ -111,6 +111,22 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
             objectKey, bucketMD, false, size, null, null, authInfo),
         log,
     };
+
+    // For Azure BlobStorage API compatability
+    // If an object already exists copy/repair creation-time
+    // creation-time must remain static after an object is created
+    //  --> EVEN FOR VERSIONS <--
+    if (objMD) {
+        if (objMD['creation-time']) {
```
Comment (jonathan-gramain) [inline lib/api/apiUtils/object/createAndStoreObject.js:129]: IMHO this is the kind of compat layer that could be done directly in `ObjectMD` class.

Comment (tmacro) [reply lib/api/apiUtils/object/createAndStoreObject.js:129]: That would be ideal, however there isn't really a place to put it where all ops touching MD would pass through. This area of the codebase is definitely in need of a refactor.

**Review by ploki — APPROVED** [review]

**Comment by tmacro** [comment]
@bert-e approve

**Review by dora-korpar — APPROVED** [review]

**Pull request merged**